src/statics.js

```diff
@@ -122,12 +122,12 @@ exports.ensure = function (path, meta) {
  * @api public
  */
 
-exports.validate = function (msg, settings) {
+exports.validate = function (msg, settings, features) {
   var all = Object.keys(this.validations)
 
   for (var i = 0, err, fn; i < all.length; ++i) {
     fn = this.validations[i].validate
-    err = fn.call(this, msg, settings)
+    err = fn.call(this, msg, settings, features)
```

Comment:
Pass in `features` to the ensure blocks:
```
.ensure(function(msg, settings, features) {
    if (features.includes('my_flag') && !settings.my_setting) {
        this.reject('new feature behind my_flag requires my_settings')
    }
}) 
```

Should the `featureOn` helper method also be supported in ensure blocks?

Reply:
We should adopt a single model for checking features across the code, especially since JS won't check if we are using the API right, it becomes easy to mistakenly use the wrong model.

Reply:
With the switch to objects we're now able to use the model in the event handlers and ensure blocks, with the exception of `this`:
```
// in ensure blocks
if (features.my_flag) {...}

// in event handlers
if (this.features.my_flag) {...}
```